Gemfile.lock

```diff
@@ -0,0 +1,366 @@
+    rspec-expectations (3.9.4)
+      diff-lcs (>= 1.2.0, < 2.0)
+      rspec-support (~> 3.9.0)
+    rspec-mocks (3.9.1)
+      diff-lcs (>= 1.2.0, < 2.0)
+      rspec-support (~> 3.9.0)
+    rspec-support (3.9.4)
+    rspec_junit_formatter (0.4.1)
+      rspec-core (>= 2, < 4, != 2.12.0)
+    rubocop (0.49.1)
+      parallel (~> 1.10)
+      parser (>= 2.3.3.1, < 3.0)
+      powerpack (~> 0.1)
+      rainbow (>= 1.99.1, < 3.0)
+      ruby-progressbar (~> 1.7)
+      unicode-display_width (~> 1.0, >= 1.0.1)
+    rubocop-require_tools (0.1.2)
+      rubocop (>= 0.49.1)
+    ruby-progressbar (1.11.0)
+    ruby2_keywords (0.0.2)
+    rubyzip (2.3.0)
+    sawyer (0.8.2)
+      addressable (>= 2.3.5)
+      faraday (> 0.8, < 2.0)
+    security (0.1.3)
+    signet (0.14.0)
+      addressable (~> 2.3)
+      faraday (>= 0.17.3, < 2.0)
+      jwt (>= 1.5, < 3.0)
+      multi_json (~> 1.10)
+    simctl (1.6.8)
+      CFPropertyList
+      naturally
+    simplecov (0.16.1)
+      docile (~> 1.1)
+      json (>= 1.8, < 3)
+      simplecov-html (~> 0.10.0)
+    simplecov-html (0.10.2)
+    sinatra (2.0.8.1)
+      mustermann (~> 1.0)
+      rack (~> 2.0)
+      rack-protection (= 2.0.8.1)
+      tilt (~> 2.0)
+    slack-notifier (2.3.2)
+    sync (0.5.0)
+    term-ansicolor (1.7.1)
+      tins (~> 1.0)
+    terminal-notifier (2.0.0)
+    terminal-table (1.8.0)
+      unicode-display_width (~> 1.1, >= 1.1.1)
+    thor (1.0.1)
+    tilt (2.0.10)
+    tins (1.26.0)
+      sync
+    tty-cursor (0.7.1)
+    tty-screen (0.8.1)
+    tty-spinner (0.9.3)
+      tty-cursor (~> 0.7)
+    uber (0.1.0)
+    unf (0.1.4)
+      unf_ext
+    unf_ext (0.0.7.7)
+    unicode-display_width (1.7.0)
+    webmock (3.11.0)
+      addressable (>= 2.3.6)
+      crack (>= 0.3.2)
+      hashdiff (>= 0.4.0, < 2.0.0)
+    word_wrap (1.0.0)
+    xcode-install (2.6.8)
+      claide (>= 0.9.1, < 1.1.0)
+      fastlane (>= 2.1.0, < 3.0.0)
+    xcodeproj (1.19.0)
+      CFPropertyList (>= 2.3.3, < 4.0)
+      atomos (~> 0.1.3)
+      claide (>= 1.0.2, < 2.0)
+      colored2 (~> 3.1)
+      nanaimo (~> 0.3.0)
+    xcov (1.4.3)
+      fastlane (>= 2.82.0, < 3.0.0)
+      multipart-post
+      slack-notifier
+      terminal-table
+      xcodeproj
+    xcpretty (0.3.0)
+      rouge (~> 2.0.7)
+    xcpretty-travis-formatter (1.0.1)
+      xcpretty (~> 0.2, >= 0.0.7)
+    yard (0.9.26)
+
+PLATFORMS
+  ruby
+
+DEPENDENCIES
+  climate_control (~> 0.2.0)
+  coveralls (~> 0.8.13)
+  danger (~> 8.0)
+  danger-junit (~> 1.0)
+  fakefs (~> 1.2.1)
+  fastlane!
+  fastlane-plugin-clubmate
+  fastlane-plugin-ruby
+  fastlane-plugin-slack_train (>= 0.2.0)
+  pry
+  pry-byebug
+  pry-rescue
+  pry-stack_explorer
+  rake
+  rb-readline
+  rest-client (>= 1.8.0)
+  rspec (~> 3.9.0)
+  rspec_junit_formatter (~> 0.4.1)
+  rubocop (= 0.49.1)
+  rubocop-require_tools (>= 0.1.2)
+  sinatra (~> 2.0.8)
+  webmock (~> 3.8)
+  xcode-install (>= 2.6.7)
+  xcov (~> 1.4.1)
+  yard (~> 0.9.11)
+
+BUNDLED WITH
+   2.2.4
```

Comment:
This doesn't mean we force end-users to use this version of `bundler`, of course.

Reply:
Sorry, it's not obvious to me how this doesn't force users to use this version of bundler 😬 could you explain? 😅 
I've had trouble with this in the past where I had a different bundler version than the one specified by a Gemfile.lock and I ended up either changing the Gemfile.lock or changing my bundler version (not sure which way I went, it was a long time ago)

Reply:
@rogerluan 

`Gemfile.lock`, in this case, is not something that affects ".gem" package to be installed at all. All needed to make ".gem" package is "gemspec" file. This is for sharing what fastlane contributors install to their own environment. Again, that way we have the consistency of behaviour in unit testing and can be likely to replicate the same bug in others' environment. (Importantly this applies to CI😄)

Regarding your issue, I suppose that the issue you had was due to `bundler`' versioning issue between the major versions; 1 and 2. For example, if you have only `bundler` **1** installed locally, `bundle install` against this lock file will fail. 

<details> 

```
% gem install bundler -v 1.17
Fetching: bundler-1.17.0.gem (100%)
Successfully installed bundler-1.17.0
Parsing documentation for bundler-1.17.0
Installing ri documentation for bundler-1.17.0
Done installing documentation for bundler after 4 seconds
1 gem installed
% bundle install
You must use Bundler 2 or greater with this lockfile.
```

</details>

This is what I saw on my testing. Only the mentioned case above fails. The differences in minor version didn't affect any for `bundle install` (`bundle` will just warn the difference in the console) 

Installed version  \ Locked version| 1.x | 2.x |
--|--|--
1.x | 🆗 | 🙅‍♂️
2.x | 🆗  ( lock file will be automatically migrated to 2.x) | 🆗  

The good thing is that `bundler` 1 and 2 can coexist so if you have both installed, you don't have to remove either one for a project using another one. 

> Version autoswitching means that Bundler will automatically run the correct 1.x or 2.x version based on your lockfile. This means you can have both Bundler 1.17.3 and Bundler 2.0.0 installed at the same time, and each application will use the version saved in your lockfile when you run bundle install.

https://bundler.io/blog/2019/01/03/announcing-bundler-2.html

I would say there are no reasons to start using `bundler` **1** since it's been two years from the release of **2**. Ruby 2.7 has got `bunder` 2.x version by default. 

https://www.ruby-lang.org/en/news/2019/12/25/ruby-2-7-0-released/

--------

Does that make sense...?

Reply:
Ah gotcha, yeah that makes total sense! 💪